manifests/set_schedule.pp

```diff
@@ -1,33 +1,156 @@
+# Sets a schedule for AIDE to run a check on your system
 #
-# Enabling this meets CCE-27222-9.
+# @param method
+#   How you wish to schedule the run
 #
-# @param minute ``minute`` cron parameter
-# @param hour ``hour`` cron parameter
-# @param monthday ``monthday`` cron parameter
-# @param month ``month`` cron parameter
-# @param weekday ``weekday`` cron parameter
-# @param command ``command`` cron parameter
+# @param systemd_calendar
+#   If `$method` is `systemd`, set this exact calendar string
+#
+#   This is not verified, use `systemd-analyze calendar` on a modern system to
+#   ensure that you have a valid string
+#
+# @param minute
+#   `minute` cron parameter
+#
+# @param hour
+#   `hour` cron parameter
+#
+# @param monthday
+#   `monthday` cron parameter
+#
+# @param month
+#   `month` cron parameter
+#
+# @param weekday
+#   `weekday` cron parameter
+#
+# @param command
+#   `command` cron parameter
 #
 # @author https://github.com/simp/pupmod-simp-aide/graphs/contributors
 #
 class aide::set_schedule (
-  Simplib::Cron::Minute   $minute   = $::aide::minute,
-  Simplib::Cron::Hour     $hour     = $::aide::hour,
-  Simplib::Cron::Monthday $monthday = $::aide::monthday,
-  Simplib::Cron::Month    $month    = $::aide::month,
-  Simplib::Cron::Weekday  $weekday  = $::aide::weekday,
-  String                  $command  = $::aide::cron_command,
+  Enum['root', 'etc', 'systemd'] $method           = pick(getvar('aide::cron_method'), 'systemd'),
+  Optional[String[1]]            $systemd_calendar = getvar('aide::systemd_calendar'),
+  Simplib::Cron::Minute          $minute           = pick(getvar('aide::minute'), fqdn_rand(59)),
+  Simplib::Cron::Hour            $hour             = pick(getvar('aide::hour'), 4),
+  Simplib::Cron::Monthday        $monthday         = pick(getvar('aide::monthday'), '*'),
+  Simplib::Cron::Month           $month            = pick(getvar('aide::month'), '*'),
+  Simplib::Cron::Weekday         $weekday          = pick(getvar('aide::weekday'), 0),
+  String                         $command          = pick(getvar('aide::cron_command'), '/bin/nice -n 19 /usr/sbin/aide --check')
 ) {
   assert_private()
 
-  cron { 'aide_schedule':
-    command  => $command,
-    user     => 'root',
-    minute   => $minute,
-    hour     => $hour,
-    monthday => $monthday,
-    month    => $month,
-    weekday  => $weekday
+  if $systemd_calendar {
+    $_systemd_calendar = $systemd_calendar
+  }
+  else {
+    $_systemd_calendar = simplib::cron::to_systemd(
+      $minute,
+      $hour,
+      $month,
+      $monthday,
+      $weekday
+    )
+  }
+
+  $_timer = @("EOM")
+  [Timer]
+  OnCalendar=$_systemd_calendar
+  EOM
+
+  $_service = @("EOM")
+  [Service]
+  Type=oneshot
+  # Exit codes that simply mean that something was changed or updated but that
+  # the check ran successfully
+  SuccessExitStatus=1 2 3 4 5 6 7
+  ExecStart=$command
+  EOM
+
+  systemd::timer { 'puppet_aide.timer':
```

Comment:
This resource is being created even when the user selects another option.  I understand that this systemd timer is disabled if not using 'systemd' cron method, but other design pattern of resource ensure absent is being used the rest of the file.  Why was this method used instead? Does it not apply to the system:timer resource?